client/src/components/Modals/SignUpModal.js

```diff
@@ -0,0 +1,114 @@
+
+class SignUpModal extends Component {
+  constructor () {
+    super()
+    this.state = {
+      name: '',
+      email: '',
+      password: '',
+      confirmPassword: '',
+      passwordError: '',
+      signUpError: ''
+    }
+  }
+
+  handleInputChange = event => {
+    this.setState({ [event.target.name]: event.target.value })
+  }
+
+  render () {
+    const { name, email, password, confirmPassword } = this.state
+    return (
+      <Modal className='signup_login_modal' isOpen={this.props.signUpModal} toggle={this.props.toggleSignUp}>
+        <ModalHeader className='signup_login_modal_header'>
+          <span>Sign up for free to create study sets</span>
+        </ModalHeader>
+        <Mutation mutation={newUserMutation}>
+          {(createNewUser, { loading }) => (
+            <form onSubmit={event => {
+              event.preventDefault()
+              if (this.state.password !== this.state.confirmPassword) {
+                this.setState({ passwordError: 'Passwords do not match.' })
+              } else {
+                const createdUser = createNewUser({ variables: { TeacherName: name, TeacherEmail: email, TeacherPW: password } })
+                createdUser.then(() => {
+                  this.setState({
+                    name: '',
+                    email: '',
+                    password: '',
+                    confirmPassword: '',
+                    passwordError: false
+                  })
+                  this.props.toggleSignUp()
+                  this.props.toggleLogIn()
+                }).catch(() => {
+                  this.setState({
+                    signUpError: 'Email Already Exists',
+                    passwordError: ''
+                  })
+                })
+              }
+            }}>
+              <ModalBody className='signup_login_modal_body'>
+                {loading ? <span>Checking information. Please wait...</span>
+                  : <div>
+                    <div className='modal_div'>
+                      <span>NAME</span>
+                      <input type='text' name='name' value={name} onChange={this.handleInputChange} required />
+                    </div>
+                    <div className='modal_div'>
+                      <span>EMAIL</span>
+                      <input type='email' name='email' value={email} onChange={this.handleInputChange} required />
+                      {this.state.signUpError ? <span className='error'>{this.state.signUpError}</span> : null}
+                    </div>
+                    <div className='modal_div'>
+                      <span>PASSWORD</span>
+                      <input type='password' name='password' value={password} onChange={this.handleInputChange} required />
+                      {this.state.passwordError ? <span className='error'>{this.state.passwordError}</span> : null}
+                    </div>
+                    <div className='modal_div'>
+                      <span>CONFIRM PASSWORD</span>
```

Comment:
This doesn't need to be changed yet, however I'm thinking it might be better to have a "show/hide password" button and skip the whole "confirm password" step entirely, apparently professional UX designers recommend doing it that way, and it looks pretty easy to accomplish.